integrationtest/pg/pg_test.go

```diff
@@ -0,0 +1,264 @@
+	require.NoError(t, err)
+	require.Equal(t, 1, len(lst))
+}
+
+func TestArchiveBucketWorkflow(t *testing.T) {
+	_ = spinup(t)
+	ctx, client := setup(t)
+
+	// Create bucket with a file.
+	b, err := client.Init(ctx, "bucky")
+	require.Nil(t, err)
+	time.Sleep(4 * time.Second) // Give a sec to fund the Fil address.
+	rootCid1 := addDataFileToBucket(ctx, t, client, b.Root.Key, "Data1.txt")
+
+	// Archive it (push to PG)
+	_, err = client.Archive(ctx, b.Root.Key)
+	require.NoError(t, err)
+
+	// Wait for the archive to finish.
+	require.Eventually(t, archiveFinalState(ctx, t, client, b.Root.Key), 60*time.Second, 2*time.Second)
+
+	// Verify that the current archive status is Done.
+	as, err := client.ArchiveStatus(ctx, b.Root.Key)
+	require.NoError(t, err)
+	require.Equal(t, buckets_pb.ArchiveStatusReply_Done, as.GetStatus())
+
+	// Get ArchiveInfo, which has all successful pushs with
+	// its data about deals.
+	ai, err := client.ArchiveInfo(ctx, b.Root.Key)
+	require.NoError(t, err)
+
+	archive := ai.GetArchive()
+	require.Equal(t, rootCid1, archive.Cid)
+	require.Len(t, archive.Deals, 1)
+	deal := archive.Deals[0]
+	require.NotEmpty(t, deal.GetProposalCid())
+	require.NotEmpty(t, deal.GetMiner())
+
+	// Add another file to the bucket.
+	rootCid2 := addDataFileToBucket(ctx, t, client, b.Root.Key, "Data2.txt")
+
+	// Archive again.
+	_, err = client.Archive(ctx, b.Root.Key)
+	require.NoError(t, err)
+	require.Eventually(t, archiveFinalState(ctx, t, client, b.Root.Key), 60*time.Second, 2*time.Second)
+	as, err = client.ArchiveStatus(ctx, b.Root.Key)
+	require.NoError(t, err)
+	require.Equal(t, buckets_pb.ArchiveStatusReply_Done, as.GetStatus())
+
+	ai, err = client.ArchiveInfo(ctx, b.Root.Key)
+	require.NoError(t, err)
+
+	archive = ai.GetArchive()
+	require.Equal(t, rootCid2, archive.Cid)
+	require.Len(t, archive.Deals, 1)
+	deal = archive.Deals[0]
+	require.NotEmpty(t, deal.GetProposalCid())
+	require.NotEmpty(t, deal.GetMiner())
+
+}
+
+func TestArchiveWatch(t *testing.T) {
+	_ = spinup(t)
+	ctx, client := setup(t)
+
+	b, err := client.Init(ctx, "bucky")
+	require.Nil(t, err)
+	time.Sleep(4 * time.Second)
+	addDataFileToBucket(ctx, t, client, b.Root.Key, "Data1.txt")
+
+	_, err = client.Archive(ctx, b.Root.Key)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	ch := make(chan string, 100)
+	go func() {
+		err = client.ArchiveWatch(ctx, b.Root.Key, ch)
+		close(ch)
+	}()
+	count := 0
+	for s := range ch {
+		require.NotEmpty(t, s)
+		count++
+		if count > 4 {
+			cancel()
+		}
+	}
+	require.NoError(t, err)
+	require.Greater(t, count, 3)
+}
+
+func TestFailingArchive(t *testing.T) {
+	_ = spinup(t)
+	ctx, client := setup(t)
+
+	b, err := client.Init(ctx, "bucky")
+	require.Nil(t, err)
+	time.Sleep(4 * time.Second)
+	// Store a file that is bigger than the sector size, this
+	// should lead to an error on the PG side.
+	addDataFileToBucket(ctx, t, client, b.Root.Key, "Data3.txt")
```

Comment:
Have you set the sector size to be small somewhere? Not sure how that all works.

Reply:
The `textileio/lotus-devnet` has a fixed 2KB sector size. I'm not sure that's tunable without breaking some other sectorbuilder-mock implementation assumptions. So you can imagine that being _baked-in_. 
But now that you mention that I could try and see what happen to maybe make that tunable if magically works.